### PR TITLE
Fail the test and stop it from activating if the simpleRate is set to one recipe but no others. For #91 issue.

### PR DESCRIPTION
**Pull request by allmywant**

### CODE REVIEW

File: lib/mojito.js [hunk 1]
```diff
@@ -57,17 +57,23 @@ Mojito = (function () {
...
 
                 if (recipe.sampleRate != null) {
                     totalSimpleRate += recipe.sampleRate;
+                    oneRecipeRateSet = true;
+                } else {
+                    oneRecipeRateNotSet = true;
                 }
             }
 
+            if (oneRecipeRateSet && oneRecipeRateNotSet) throw new Error('The sampleRate property is set to one recipe object, but not others');
```
Comment (dapperdrop) [inline lib/mojito.js]: Thinking of a more descriptive error message with suggested fix, e.g.
"Recipe-level sampleRate property must be defined for all recipes if it is being used"

Thoughts @kingo55?

Comment (kingo55) [reply lib/mojito.js]: Yes, I think so.

Comment (dapperdrop) [reply lib/mojito.js]: Cool, @allmywant - please change the error string to: "Recipe-level sampleRate property must be defined for all recipes if it is being used" and we should be right to approve + merge.

File: tests/test_suite.html [hunk 2]
```diff
@@ -548,7 +548,8 @@
             },
             recipes: {
                 '0': {
-                    name: 'control'
+                    name: 'control',
+                    sampleRate: 0
```
Comment (kingo55) [inline tests/test_suite.html:552]: @allmywant @dapperdrop - Should we introduce a test for the recipe sample rates failure?

File: lib/mojito.js (hunk 1 above)
Comment (allmywant) [reply lib/mojito.js]: @dapperdrop 
Error message updated.

File: tests/test_suite.html (hunk 2 above)
Comment (dapperdrop) [reply tests/test_suite.html:552]: Yes I think we should.

Comment (allmywant) [reply tests/test_suite.html:552]: @kingo55 
Yes we can. But we must wrap the code with try/catch or gulp test will stop running the subsequent tests.

Comment (kingo55) [reply tests/test_suite.html:552]: OK - good point @allmywant. Please add a test inside try..catch.

Comment (allmywant) [reply tests/test_suite.html:552]: @kingo55 
Added a test case for this.

File: tests/test_suite.html [hunk 3]
```diff
@@ -743,6 +743,34 @@
...
+                    '1': {
+                        name: 'treatment'
+                    }
+                }
+            });
+        } catch(e) {
+            it('Recipe-level sampleRate property must be defined for all recipes if it is being used', function () {
+                w16Activated.should.equal(false);
+            });
+        }
```
Comment (kingo55) [inline tests/test_suite.html]: If no error is thrown, I think this unit test is never run. e.g. if the w16Activated variable is set to true and no error is thrown, the catch statement is not evaluated:

Screen Shot 2021-03-18 at 18 22 29

Should it sit outside the try..catch statement?

Comment (kingo55) [reply tests/test_suite.html]: Oh and while we're there, let's give the test a name like:

`Tests fail when recipe-level sample rates are set for only some recipes.`

Comment (allmywant) [reply tests/test_suite.html]: Thanks @kingo55 , you're right, it should sit outside the  try..catch block. 
Fixed and updated the test name.

Comment (kingo55) [reply tests/test_suite.html]: Awesome... Thanks @allmywant - looking good now!

Want to do the honours and merge/squash?

Comment (allmywant) [reply tests/test_suite.html]: @kingo55 
Merged. It seemed I should do `git merge squash` first, I remember you shared an article about it but I forgot the URL, can you please share it again?

**Pull request merged**